### PR TITLE
External Postgres upgrade doc

### PR DESCRIPTION
**Pull request by shivangi-singh1**
Signed-off-by: Shivangi Singh <shisingh@progress.com>

### :+1: Definition of Done
Doc on Automate Postgres upgrade from v9.6 to v13 has to be made mentioning steps to upgrade external postgres to version 13.

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
++++
+title = "Automate External Postgres Upgrade from v9.6 to v13"
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:2]:
```suggestion
title = "Automate External PostgreSQL Upgrade from v9.6 to v13"
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+date = 2018-03-26T15:27:52-07:00
+draft = false
+
+gh_repo = "automate"
+[menu]
+  [menu.automate]
+    title = "External Postgres Upgrade"
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:10]:
```suggestion
    title = "External PostgreSQL Upgrade"
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    parent = "automate/configuring_automate"
+    identifier = "automate/configuring_automate/postgres_upgrade.md Automate External Postgres Upgrade"
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:12]:
```suggestion
    identifier = "automate/configuring_automate/postgres_upgrade.md Automate External PostgreSQL Upgrade"
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    weight = 10
++++
+
+## Upgrade path for Chef Automate users on Postgres v9.6
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:16]:
```suggestion
## Upgrade path for Chef Automate users on PostgreSQL v9.6
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+1. Take backup of chef automate - [Backup]({{< ref "backup.md" >}})
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:17]:
```suggestion
1. Take backup of Chef Automate - [Backup]({{< ref "backup.md" >}})
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+   * This is necessary just to ensure in case of any failure, there is a state to return to.
+
+   * On Chef Automate machine
+
+   * For Airgapped machine please follow [Airgapped Installation]({{< ref "airgapped_installation.md" >}})  to upgrade to latest version of Chef Automate which supports External Postgres v13.
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:23]:
```suggestion
   * For Airgapped machine please follow [Airgapped Installation]({{< ref "airgapped_installation.md" >}})  to upgrade to the latest version of Chef Automate which supports External PostgreSQL v13.
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+   * For Non-Airgapped machine, Upgrade Chef Automate to latest version which supports External Postgresql v13:
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:25]:
```suggestion
   * For Non-Airgapped machine, Upgrade Chef Automate to the latest version which supports External PostgreSQL v13:
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+   * After successful upgrade
+
+   * Stop Chef Automate Services
+
+    ```shell
+    sudo chef-automate stop
+    ```
+
+3. Upgrade Postgesql 9.6 to 13.4
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:39]:
```suggestion
3. Upgrade PostgreSQL 9.6 to 13.4
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+   * On External Postgres, please contact your database administrator to do the upgrade
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:41]:
```suggestion
   * On External PostgreSQL, please contact your database administrator to perform the upgrade
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+   * Check all the services are up and running in Automate
+
+    ```shell
+    sudo chef-automate status
+    ```
+
+
+## Upgrade Failure Troubleshooting
+
+1. If the upgrade failed and you are left with a corrupted Chef Automate and/or a corrupted database, **DO NOT RISK YOUR BACKUP OF AUTOMATE**. Take all steps necessary to preserve the backup, including copying it to another disk. Consult with a professional sysadmin for instructions and best practices.
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:62]:
```suggestion
1. If the upgrade failed and you are left with a corrupted Chef Automate and/or a corrupted database, **DO NOT RISK YOUR BACKUP OF AUTOMATE**. Take all steps necessary to preserve the backup, including copying it to another disk.
```

Anyone reading this a professional sysadmin

File: components/docs-chef-io/content/automate/postgres_upgrade.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+2. Contact customer support.
```
Comment (tas50) [inline components/docs-chef-io/content/automate/postgres_upgrade.md:64]:
```suggestion
2. Contact Chef customer support.
```

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 1 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 2 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 3 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 4 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md:16]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 5 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md:17]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 6 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md:23]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 7 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md:25]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 8 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md:39]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 9 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md:41]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 10 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md:62]: Done

File: components/docs-chef-io/content/automate/postgres_upgrade.md (hunk 11 above)
Comment (shivangi-singh1) [reply components/docs-chef-io/content/automate/postgres_upgrade.md:64]: Done

File: components/docs-chef-io/content/automate/external_postgres_upgrade_steps.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+    ```
+3. If the PostgreSQL v9.6 config was modified or customized. Please check the differences and update the new PostgreSQL v13 config with similar changes. Please connect with your database administrator if you don't know these changes.
+    ```shell
+    sudo sdiff -s /etc/postgresql/9.6/main/postgresql.conf /etc/postgresql/13/main/postgresql.conf
+    sudo sdiff -s /etc/postgresql/9.6/main/pg_hba.conf /etc/postgresql/13/main/pg_hba.conf
+    ```
+4. Stop Both PostgreSQL Servers:
+    ```shell
+    sudo systemctl stop postgresql.service
+    ```
```
Comment (vivekshankar1) [inline components/docs-chef-io/content/automate/external_postgres_upgrade_steps.md:39]: This step has to be done after vacuuming. We should move it 2 steps below

**Review by vivekshankar1 — CHANGES_REQUESTED** [review]

**Comment by kagarmoe** [comment]
Use #6509 instead

**Comment by kagarmoe** [comment]
Closed by #6509

**Pull request closed without merging**